src/common/snippets/src/lowered/pass/fuse_loops.cpp

```diff
@@ -265,10 +266,10 @@ bool FuseLoops::run(LinearIR& linear_ir, lowered::LinearIR::constExprIt begin, l
                 // Loop_0 (Current)    Loop_0 + Loop_1 => new `Loop_0`
                 //   |               =>           |
                 // Loop_1 (Lower)                 |
-                const auto& output_ports = current_loop_info->get_output_ports();
                 bool was_fusion_down = false;
-                for (size_t out_port = 0; out_port < output_ports.size() && !was_fusion_down; ++out_port) {
-                    const auto output_port = output_ports[out_port];
+                const auto output_port_count = current_loop_info->get_output_count();
+                for (size_t out_port = 0; !was_fusion_down && out_port < output_port_count; ++out_port) {
+                    const auto output_port = current_loop_info->get_output_ports()[out_port];
```

Comment:
It looks a bit expensive to `call get_output_ports()` on every iteration, since the whole vector will be coipied every time.
Why can't we copy all the output ports once before the loop the same way we handle input ports on L224?

Reply:
We can do as for input ports as well. Why not.

Moreover, we can save them by reference since we need to them only in loop condition where `!was_fusion_down` is first check now (`!was_fusion_down = false`  only when `current_loop_info` has been changed)